src/basho_bench_stats.erl

```diff
@@ -333,7 +332,59 @@ consume_report_msgs() ->
             ok
     end.
 
+
 % Assuming all stats sink modules are prefixed with basho_bench_stats_writer_
 normalize_name(StatsSink) when is_atom(StatsSink) ->
     {ok, list_to_atom("basho_bench_stats_writer_" ++ atom_to_list(StatsSink))};
 normalize_name(StatsSink) -> {error, {StatsSink, invalid_name}}.
+
+
+% Build up the list of operations taking into account the use of worker groups
+% If workers are not used, use the global operations list
+% If workers are being used, use worker_type as an operation prefix,
+%    using either global or per-worker operations depending on the situation.
+%
+
+get_worker_ops() ->
+   Workers = basho_bench_config:get(workers, []),
+   case Workers of
+       %% No workers reverts to original case is fine as long as we stick with 2-tuples
+       [] -> 
+            lists:map(
+                fun({OpTag, _Count}) -> {OpTag, OpTag};
+                    ({Label, OpTag, _Count}) -> {Label, OpTag};
+                    ({Label, OpTag, _Count, _OptionsList}) -> {Label, OpTag}
+                end, basho_bench_config:get(operations, []));
+       %% Use workers to determine active worker types for current config
+        _ -> 
+            get_worker_ops(Workers, basho_bench_config:get(worker_types), [])
+    end.
+
+
+get_worker_ops([], _WorkerTypes, Acc) ->
+     Acc;
+get_worker_ops(Workers, WorkerTypes, Acc) ->
+     [WorkerEntry | RestWorkers] = Workers,
+     {WorkerType, _Count} = WorkerEntry,
+     WorkerConfig = proplists:get_value(WorkerType, WorkerTypes),
+     %% Get either per-worker ops or global ops
+     WorkerOps = proplists:get_value(operations, WorkerConfig),
+
+     Ops = lists:map(
+        fun({OpTag, _Count2}) -> 
+                { worker_op_name(WorkerType, OpTag), worker_op_name(WorkerType, OpTag)};
+           ({Label, OpTag, _Count2}) ->
+                { worker_op_name(WorkerType, Label), worker_op_name(WorkerType, OpTag)};
+           ({Label, OpTag, _Count2, _OptionsList}) ->
+                { worker_op_name(WorkerType, Label), worker_op_name(WorkerType, OpTag)}
```

Comment:
That was what I did, I yanked it. It was written to try to support per-operation parameters but that would run against the basic mechanism of basho_bench_config:get retrieving values from the process_dict or global environment to work. 

If it is mainly keygen and valgen that we need, we now have both per-worker config and api-state based named keygens to provide a lot more functionality than we had. I filed FB 84560 with the code that I yanked out so we didn't loose it but it wasn't working as implemented. If we need it it would be helpful to have some examples to clarify how it needs to work.